flask_security/decorators.py

```diff
@@ -53,7 +53,11 @@ def _get_unauthorized_view():
             except BuildError:
                 view = None
         utils.do_flash(*utils.get_message('UNAUTHORIZED'))
-        return redirect(view or request.referrer or '/')
+        redirect_to = '/'
+        if request.referrer and not request.referrer.endswith(request.path):
```

Comment:
@nfvs you can check the results of the tests if you revert `flask_security/decorators.py`. I am usually checking the official docs on http://flask.pocoo.org/docs/0.12/api/#flask.Request